### PR TITLE
ignore executor does not exist when removing

### PR DESCRIPTION
**Pull request by maxkozlovsky**
This can happen if the extractor retries unregister request, ignore does not exist error.

### CODE REVIEW

File: src/state/store/state_machine_objects.rs [hunk 1]
```diff
@@ -200,6 +200,13 @@ impl ExtractorExecutorsTable {
             .remove(executor_id);
     }
 
+    pub fn remove_by_executor(&self, executor_id: &ExecutorId) {
```
Comment (diptanu) [inline src/state/store/state_machine_objects.rs:203]: Add some comments here?

File: src/state/store/state_machine_objects.rs [hunk 2]
```diff
@@ -1252,8 +1260,12 @@ impl IndexifyState {
                     .map_err(|e| StateMachineError::TransactionError(e.to_string()))?;
 
                 //  Remove the the extractor from the executor -> extractor mapping table
-                self.extractor_executors_table
-                    .remove(&executor_meta.extractor.name, &executor_meta.id);
+                if let Some(executor_meta) = executor_meta {
+                    self.extractor_executors_table
+                        .remove(&executor_meta.extractor.name, &executor_meta.id);
+                } else {
+                    self.extractor_executors_table.remove_by_executor(executor_id);
```
Comment (diptanu) [inline src/state/store/state_machine_objects.rs:1267]: why is this necessary? A single executor belongs to only one extractor right now. When do we need to remove the executor entry from all the extractor -> executor table?

**Review by diptanu — CHANGES_REQUESTED** [review]
Fix the tests and please add comments.

File: src/state/store/state_machine_objects.rs (hunk 2 above)
Comment (maxkozlovsky) [reply src/state/store/state_machine_objects.rs]: Ok removed trying remove by id
